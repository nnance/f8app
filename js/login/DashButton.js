import React from 'React';
import { View, Text, TouchableOpacity } from 'react-native';

import F8Colors from 'F8Colors';

/* eslint react/no-multi-comp: off */

export default class DashButton extends React.Component {
  render() {
    return (
      <TouchableOpacity onPress={this.props.onPress} style={{ flexDirection: 'row', padding: 10 }}>
        <View
          style={{
            height: 1,
            flex: 1,
            marginVertical: 12,
            marginHorizontal: 15,
            backgroundColor: F8Colors.fadedLightText,
          }}
        />
        <Text
          style={{ color: 'white' }}
        >
          {this.props.caption}
        </Text>
        <View
          style={{
            height: 1,
            flex: 1,
            marginVertical: 12,
            marginHorizontal: 15,
            backgroundColor: F8Colors.fadedLightText,
          }}
        />
      </TouchableOpacity>
    );
  }
}

export class DashButtonWithContainer extends React.Component {
  render() {
    const props = this.props;
    return (<View style={[{ flex: 1, justifyContent: 'flex-end', alignItems: 'center' }, props.style]}>
      <DashButton {...props} />
    </View>);
  }
}
