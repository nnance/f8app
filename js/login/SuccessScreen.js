

import React from 'React';
import { Image, View, Text } from 'react-native';

import styles from './styles';

export default ({ goToLogin, successText }) => (
  <Image
    style={styles.container}
    source={require('../assets/login/email-bg.png')}
  >
    <View style={styles.inputSession}>
      <Text style={styles.successText} onPress={goToLogin}>{successText}, go back to login</Text>
    </View>
  </Image>
);
