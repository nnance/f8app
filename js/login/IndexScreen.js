/**
 * Copyright 2016 Facebook, Inc.
 *
 * You are hereby granted a non-exclusive, worldwide, royalty-free license to
 * use, copy, modify, and distribute this software in source code or binary
 * form for use in connection with the web services and APIs provided by
 * Facebook.
 *
 * As with any software that integrates with the Facebook platform, your use
 * of this software is subject to the Facebook Developer Principles and
 * Policies [http://developers.facebook.com/policy/]. This copyright notice
 * shall be included in all copies or substantial portions of the software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 * DEALINGS IN THE SOFTWARE
 *
 * @flow
 */


import ClogiiButton from 'ClogiiButton';
import {
  StatusBar,
  View,
  Image,
} from 'react-native';

import { DashButtonWithContainer } from './DashButton';

const F8Colors = require('F8Colors');
const React = require('React');
const StyleSheet = require('StyleSheet');
const { Text } = require('F8Text');
const { connect } = require('react-redux');
const LoginButton = require('../common/LoginButton');

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'transparent',
    padding: 26,
    // Image's source contains explicit size, but we want
    // it to prefer flex: 1
    width: undefined,
    height: undefined,
  },
  section: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  last: {
    justifyContent: 'flex-end',
  },
  button: {
    margin: 10,
    height: 40,
    width: 250,
  },
  loginComment: {
    marginBottom: 14,
    fontSize: 12,
    color: F8Colors.darkText,
    textAlign: 'center',
  },
  skip: {
    position: 'absolute',
    right: 0,
    top: 20,
    padding: 15,
  },
});

class IndexScreen extends React.Component {
  componentDidMount() {
    if (StatusBar) {
      StatusBar.setBarStyle('default');
    }
  }

  render() {
    return (
      <Image
        style={styles.container}
        source={require('../assets/login/bg.png')}
      >
        <View style={{ flex: 7, alignItems: 'center', justifyContent: 'center' }}>
          <View style={{ flex: 4 }} />
          <View style={{ flex: 3, flexDirection: 'row', justifyContent: 'center', alignItems: 'center' }}>
            <View style={{ flex: 1 }}>
              <Image
                style={{ flex: 1, resizeMode: 'contain', width: undefined, height: undefined }}
                source={require('../assets/login/header.png')}
              />
            </View>
          </View>
          <Text style={{ flex: 4, fontSize: 12, color: 'white' }}>ที่ที่จักวาลของคุณ จะอยู่ง่าย... แค่ปลายนิ้ว</Text>
        </View>
        <View style={{ flex: 4, alignItems: 'center', justifyContent: 'center' }}>
          <LoginButton source="First screen" style={styles.button} />
          <ClogiiButton
            type="bordered"
            style={styles.button}
            onPress={() => this.props.pushPage('email-login')}
            caption="ลงชื่อเข้าใช้ผ่านอีเมล"
          />
          <DashButtonWithContainer caption="สร้างบัญชีใหม่" onPress={() => this.props.pushPage('signup')} />
        </View>
        <View style={{ flex: 1, alignItems: 'center', justifyContent: 'flex-end' }}>
          <Text style={{ color: F8Colors.fadedLightText, fontSize: 10 }}>
            By signing up you agree to our Terms & Privacy policy
          </Text>
        </View>
      </Image>
    );
  }
}

export default connect()(IndexScreen);
export {
  IndexScreen as Component,
};
