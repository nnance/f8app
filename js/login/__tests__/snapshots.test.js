import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import DashButton, { DashButtonWithContainer } from '../DashButton';
import SuccessScreen from '../SuccessScreen';
import { Component as IndexScreenComponent } from '../IndexScreen';

describe('DashButton', () => {
  it('render', () => {
    const tree = renderer.create(<DashButton caption="test" />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render with container', () => {
    const tree = renderer.create(<DashButtonWithContainer caption="test container" />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('SuccessScreen', () => {
  it('render', () => {
    const tree = renderer.create(<SuccessScreen successText="test" />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('IndexScreen', () => {
  it('render', () => {
    const tree = shallow(<IndexScreenComponent pushPage={jest.fn()} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
