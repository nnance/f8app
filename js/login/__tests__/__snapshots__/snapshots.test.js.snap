exports[`DashButton render 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "opacity": 1,
      "padding": 10,
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.5)",
        "flex": 1,
        "height": 1,
        "marginHorizontal": 15,
        "marginVertical": 12,
      }
    } />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
      }
    }>
    test
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.5)",
        "flex": 1,
        "height": 1,
        "marginHorizontal": 15,
        "marginVertical": 12,
      }
    } />
</View>
`;

exports[`DashButton render with container 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
      },
      undefined,
    ]
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "opacity": 1,
        "padding": 10,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.5)",
          "flex": 1,
          "height": 1,
          "marginHorizontal": 15,
          "marginVertical": 12,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }>
      test container
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.5)",
          "flex": 1,
          "height": 1,
          "marginHorizontal": 15,
          "marginVertical": 12,
        }
      } />
  </View>
</View>
`;

exports[`IndexScreen render 1`] = `
<Image
  source={1}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
      "height": undefined,
      "padding": 26,
      "width": undefined,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 7,
        "justifyContent": "center",
      }
    }>
    <View
      style={
        Object {
          "flex": 4,
        }
      } />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }>
      <View
        style={
          Object {
            "flex": 1,
          }
        }>
        <Image
          source={1}
          style={
            Object {
              "flex": 1,
              "height": undefined,
              "resizeMode": "contain",
              "width": undefined,
            }
          } />
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "white",
          "flex": 4,
          "fontSize": 12,
        }
      }>
      ที่ที่จักวาลของคุณ จะอยู่ง่าย... แค่ปลายนิ้ว
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 4,
        "justifyContent": "center",
      }
    }>
    <Connect(LoginButton)
      source="First screen"
      style={
        Object {
          "height": 40,
          "margin": 10,
          "width": 250,
        }
      } />
    <ClogiiButton
      caption="ลงชื่อเข้าใช้ผ่านอีเมล"
      onPress={[Function]}
      style={
        Object {
          "height": 40,
          "margin": 10,
          "width": 250,
        }
      }
      type="bordered" />
    <DashButtonWithContainer
      caption="สร้างบัญชีใหม่"
      onPress={[Function]} />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }>
    <Text
      style={
        Object {
          "color": "rgba(255, 255, 255, 0.5)",
          "fontSize": 10,
        }
      }>
      By signing up you agree to our Terms & Privacy policy
    </Text>
  </View>
</Image>
`;

exports[`SuccessScreen render 1`] = `
<Image
  source={1}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
      "height": undefined,
      "width": undefined,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingTop": 133.4,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={undefined}
      style={
        Object {
          "color": "white",
          "fontSize": 20,
        }
      }>
      test
      , go back to login
    </Text>
  </View>
</Image>
`;
